diff --git a/src/tint/cmd/tint/BUILD.cmake b/src/tint/cmd/tint/BUILD.cmake
index 5e7d586386..03d4b3c384 100644
--- a/src/tint/cmd/tint/BUILD.cmake
+++ b/src/tint/cmd/tint/BUILD.cmake
@@ -52,6 +52,7 @@ tint_target_add_dependencies(tint_cmd_tint_cmd cmd
   tint_lang_core_type
   tint_lang_hlsl_writer_common
   tint_lang_hlsl_writer_helpers
+  tint_lang_hlsl_writer_printer
   tint_lang_wgsl
   tint_lang_wgsl_ast
   tint_lang_wgsl_ast_transform
